test/cmake/add_subdirectories.cmake

```diff
@@ -0,0 +1,25 @@
+# -----------------------------------------------------------------------------------------------------
+# Copyright (c) 2006-2020, Knut Reinert & Freie Universität Berlin
+# Copyright (c) 2016-2020, Knut Reinert & MPI für molekulare Genetik
+# This file may be used, modified and/or redistributed under the terms of the 3-clause BSD-License
+# shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE.md
+# -----------------------------------------------------------------------------------------------------
+
+macro (add_subdirectories_of directory)
```

Comment:
Can you document input and what happens inside of the function?
Whether or not variables are exposed to the parent directory or are only valid in the subdirectories and these kind of cmake stuff? I know it is cumbersome, but might help in the future.

Reply:
:unamused: A pull request that just moved existing code into own files, all right lets make this proper documented :shrug: